canvas/image.go

```diff
@@ -285,6 +285,24 @@ func (i *Image) calculateMinSize() error {
 	return nil
 }
 
+func (img *Image) scaleSizeForCanvas(width, height int) (fyne.Size, error) {
```

Comment:
Wouldn't this be a good candidate for the new `scale` package?

Reply:
Indeed, good idea!